typeagent/aitools/embeddings.py

```diff
@@ -61,12 +61,16 @@ def __init__(
         if embedding_size is None or embedding_size <= 0:
             embedding_size = DEFAULT_EMBEDDING_SIZE
         self.embedding_size = embedding_size
-
-        if not endpoint_envvar:
-            raise ValueError(
-                f"Model {model_name} is not supported. "
-                f"Supported models are: {', '.join(model_to_embedding_size_and_envvar.keys())}"
-            )
+        
+        if required_endpoint_envvar is not None:
+            if endpoint_envvar is not None and endpoint_envvar != required_endpoint_envvar:
+                raise ValueError(
+                    f"Environment variable for endpoint {endpoint_envvar} does not match "
+                    f"required environment variable {required_endpoint_envvar} for model {model_name}."
```

Comment:
Maybe mention that this is for embeddings? E.g. (maybe overkill):

```suggestion
                    f"Environment variable for embedding endpoint {endpoint_envvar} does not match "
                    f"required environment variable {required_endpoint_envvar} for embedding model {model_name}."
```